### PR TITLE
Prevent variable leak between test

### PR DESCRIPTION
**Pull request by ph**
When running this test with a newer version of rspec we got the warning
that some values where leaking between example:

```
 1) LogStash::ShutdownWatcher when pipeline is not stalled .unsafe_shutdown = false shouldn't force the shutdown
     Failure/Error: m.call(*args)
       The use of doubles or partial doubles from rspec-mocks outside of the per-test lifecycle is not supported.
     # ./logstash-core/lib/logstash/shutdown_watcher.rb:67:in `pipeline_report_snapshot'
     # ./logstash-core/spec/logstash/shutdown_watcher_spec.rb:22:in `block in /Users/ph/es/second/logstash/logstash-core/spec/logstash/shutdown_watcher_spec.rb'
     # ./logstash-core/lib/logstash/shutdown_watcher.rb:44:in `block in start'
     # ./vendor/bundle/jruby/2.3.0/gems/stud-0.0.22/lib/stud/interval.rb:20:in `interval'
     # ./logstash-core/lib/logstash/shutdown_watcher.rb:42:in `start'
     # ./logstash-core/spec/logstash/shutdown_watcher_spec.rb:93:in `block in (root)'
```

We have added a few helpers method in the class that allowed us to
simplify the expectation and removed any usage of `Thread#kill`

### CODE REVIEW

File: logstash-core/spec/logstash/shutdown_watcher_spec.rb [hunk 1]
```diff
@@ -69,8 +59,9 @@
       it "shouldn't force the shutdown" do
         expect(subject).to_not receive(:force_exit)
         thread = Thread.new(subject) {|subject| subject.start }
-        sleep 0.1 until report_count > check_threshold
-        thread.kill
+        sleep 0.1 until subject.attempts_count > check_threshold
+        subject.stop!
+        thread.join
```
Comment (original-brownbear) [inline logstash-core/spec/logstash/shutdown_watcher_spec.rb]: @ph we may wanna join with a timeout here right, given how much fun a thread dump is from JRuby right and assert that it worked?

**Comment by original-brownbear** [comment]
@ph LGTM, just one suggestion to avoid hanging tests :)

File: logstash-core/spec/logstash/shutdown_watcher_spec.rb (hunk 1 above)
Comment (ph) [inline logstash-core/spec/logstash/shutdown_watcher_spec.rb]: updated with your suggestion.

Comment (original-brownbear) [inline logstash-core/spec/logstash/shutdown_watcher_spec.rb]: @ph nice thanks!

**Comment by original-brownbear** [comment]
@ph LGTM!

**Pull request closed without merging**